Kernel/ACPI/ACPIStaticParser.cpp

```diff
@@ -0,0 +1,385 @@
+        }
+    }
+    ASSERT_NOT_REACHED(); // If we reached this point, there's something very incorrect with the system!
+}
+
+void ACPIStaticParser::init_fadt()
+{
+    kprintf("ACPI: Initializing Fixed ACPI data\n");
+    ACPI_RAW::SDTHeader& physical_fadt = find_fadt();
+
+    // FIXME: Remove Region::Access::Write
+    OwnPtr<Region> checkup_region = MM.allocate_kernel_region((PAGE_SIZE * 2), "ACPI Static Parser", Region::Access::Read | Region::Access::Write);
+#ifdef ACPI_DEBUG
+    kprintf("ACPI: Checking FADT Length to choose the correct mapping size\n");
+#endif
+    mmap_region(*checkup_region, PhysicalAddress((u32)(&physical_fadt) & 0xfffff000));
+
+    ACPI_RAW::SDTHeader* sdt = (ACPI_RAW::SDTHeader*)(checkup_region->vaddr().get() + ((u32)(&physical_fadt) & 0xfff));
+#ifdef ACPI_DEBUG
+    kprintf("ACPI: FADT @ V 0x%x, P 0x%x\n", sdt, &physical_fadt);
+#endif
+    u32 length = sdt->length;
+    kprintf("ACPI: Fixed ACPI data, Revision %u\n", sdt->revision);
+
+    // FIXME: Remove Region::Access::Write
+    OwnPtr<Region> fadt_region = MM.allocate_kernel_region(PAGE_ROUND_UP(length) + PAGE_SIZE, "ACPI Static Parser", Region::Access::Read | Region::Access::Write);
+    mmap_region(*fadt_region, PhysicalAddress((u32)(&physical_fadt) & 0xfffff000)); // FIXME: Use bla bla & ~(PAGE_SIZE-1) instead of & 0xfffff000
+    m_fadt = new ACPI::FixedACPIData(*((ACPI_RAW::FADT*)(fadt_region->vaddr().get() + ((u32)(&physical_fadt) & 0xfff))));
+#ifdef ACPI_DEBUG
+    kprintf("ACPI: Finished to initialize Fixed ACPI data\n");
+#endif
+}
+
+void ACPIStaticParser::do_acpi_reboot()
+{
+    // FIXME: Determine if we need to do MMIO/PCI/IO access to reboot, according to ACPI spec 6.2, Section 4.8.3.6
+#ifdef ACPI_DEBUG
+    kprintf("ACPI: Rebooting, Probing FADT (@ 0x%x)\n", m_fadt.ptr());
+#endif
+    if (m_fadt->m_revision >= 2) {
+        kprintf("ACPI: Reboot, Sending value 0%x to Port 0x%x\n", m_fadt->m_reset_value, m_fadt->m_reset_reg.address);
+        IO::out8(m_fadt->m_reset_reg.address, m_fadt->m_reset_value);
+    } else {
+        kprintf("ACPI: Reboot, Not supported!\n");
+    }
+
+    ASSERT_NOT_REACHED(); /// If rebooting didn't work, halt.
+}
+
+void ACPIStaticParser::do_acpi_shutdown()
+{
+    kprintf("ACPI: Shutdown is not supported with the current configuration, Abort!\n");
+    ASSERT_NOT_REACHED();
+}
+
+void ACPIStaticParser::initialize_main_system_description_table()
+{
+    // FIXME: Remove Region::Access::Write
+    OwnPtr<Region> checkup_region = MM.allocate_kernel_region((PAGE_SIZE * 2), "ACPI Static Parser Copying Method", Region::Access::Read | Region::Access::Write);
+#ifdef ACPI_DEBUG
+    kprintf("ACPI: Checking Main SDT Length to choose the correct mapping size");
+#endif
+    mmap_region(*checkup_region, PhysicalAddress((u32)m_main_system_description_table.ptr() & 0xfffff000));
+
+    ACPI_RAW::SDTHeader* sdt = (ACPI_RAW::SDTHeader*)(checkup_region->vaddr().get() + ((u32)m_main_system_description_table.ptr() & 0xfff));
```

Comment:
Instead of repeating the type twice, please use `auto*`:
```suggestion
    auto* sdt = (ACPI_RAW::SDTHeader*)(checkup_region->vaddr().get() + ((u32)m_main_system_description_table.ptr() & 0xfff));
```